tools/backup_analysis.py

```diff
@@ -0,0 +1,175 @@
+        buf1 = fp1.read(cmp_size)
+        buf2 = fp2.read(cmp_size)
+        # If they're different, gather information.
+        if buf1 != buf2:
+            total_bytes_diff += cmp_size
+            # Count how many granularity level blocks changed.
+            if bytes_gran == 0:
+                gran_blocks += 1
+            bytes_gran += cmp_size
+    # Account for anything from the final iteration of the loop and partial blocks.
+    if partial != 0:
+        buf1 = fp1.read(partial)
+        buf2 = fp2.read(partial)
+        # If they're different, gather information.
+        if buf1 != buf2:
+            total_bytes_diff += partial
+            bytes_gran += partial
+    fp1.close()
+    fp2.close()
+    end = time.asctime()
+    if bytes_gran != 0:
+        if offset != 0:
+            print(f'{filename}: offset {start_off}: {bytes_gran} bytes differ in {granularity} bytes')
+        if bytes_gran <= pct20:
+            pct20_count += 1
+        elif bytes_gran >= pct80:
+            pct80_count += 1
+
+    # Report for each file.
+    print(f'{filename}: time: started {start} completed {end}')
+    print(f'{filename}: filesize: {file_size} differs by {total_bytes_diff} bytes in {gran_blocks} granularity blocks')
+    if gran_blocks != 0:
+        pct20_blocks = round(abs(pct20_count / gran_blocks * 100))
+        pct80_blocks = round(abs(pct80_count / gran_blocks * 100))
+        print(f'{filename}: smallest 20%: {pct20_count} of {gran_blocks} blocks ({pct20_blocks}%) differ by {pct20} bytes or less of {granularity}')
+        print(f'{filename}: largest 80%: {pct80_count} of {gran_blocks} blocks ({pct80_blocks}%) differ by {pct80} bytes or more of {granularity}')
+
+def compare_backups(dir1, dir2, granularity):
+    common=set()
+    files1=set(fnmatch.filter(os.listdir(dir1), "*.wt"))
+    files2=set(fnmatch.filter(os.listdir(dir2), "*.wt"))
+
+    common = files1.intersection(files2)
+    # For now assume the first directory is the older one. Once we add functionality to parse the
+    # text in WiredTiger.backup we can look at the checkpoint timestamp of a known table like the
+    # history store and figure out which is older.
+    for file in files1.difference(files2):
+        print(file + " dropped between backups")
+    for file in files2.difference(files1):
+        print(file + " created between backups")
+    print(common)
+    for f in common:
+        f1_size = os.stat(os.path.join(dir1, f)).st_size
+        f2_size = os.stat(os.path.join(dir2, f)).st_size
+        min_size = min(f1_size, f2_size)
+        # For now we're only concerned with changed blocks between backups.
+        # So only compare the minimum size both files have in common.
+        # FIXME: More could be done here to report extra blocks added/removed.
+        compare_file(dir1, dir2, f, min_size, 4096, granularity)
+
+def backup_analysis(args):
+    if len(args) < 2:
```

Comment:
No need to do this now, but if we ever need to add more arguments, then switching to something like `argparse` would be nice.

Reply:
Agreed. I think future arguments will likely be needed but it isn't clear what they are at the moment.